### DISCUSSION TITLE
How to display the progress of the download (clone_from) ?

### DISCUSSION THREAD

**Discussion by qcha41** [gitpython-developers/GitPython#1134] [Q&A] [answered]
Hello, 

I am a new GitPython user and I would like to display a progress bar when cloning locally a large GitHub repository.  To begin with, I have created a RemoteProgress based class and I have plugged it in the clone_from() function : 

```python
from git import Repo
from git.remote import RemoteProgress

class Progress(RemoteProgress):
    def update(self, *args):
        print(self._cur_line)

Repo.clone_from(github_url,local_path,progress=Progress())
```

When I run it, I have well the progress from 0 to 100% of _remote: Counting objects:_ and _remote: Compressing objects:_ steps (which are super fast) but after that, nothing more : the terminal keep freezed until the end of the download. 

How to catch the progress of the download itself ? 

Thank you in advance!

**Comment by Byron** [comment, marked as answer]
From what I could gather [in the docs](https://gitpython.readthedocs.io/en/stable/tutorial.html#handling-remotes) this is essentially what needs to be done, and that should include local progress (indexing, resolving) along with remote progress.

Maybe for some reason progress cannot be parsed, maybe override `line_dropped(self, line)` to learn about lines that couldn't be parsed.

Nonetheless, I would also expect that to work in a more straightforward fashion, maybe we can figure that out here.

**Comment by emcek** [comment]
Looks like similar problem... https://stackoverflow.com/questions/51045540/python-progress-bar-for-git-clone
